src/java.desktop/share/classes/sun/swing/AccessibleComponentAccessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021, JetBrains s.r.o.. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package sun.swing;
+
+import javax.accessibility.Accessible;
+import javax.accessibility.AccessibleContext;
+import java.util.List;
+import java.util.function.Function;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class AccessibleComponentAccessor {
+    private static final List<Function<AccessibleContext, Object>> accessors = new CopyOnWriteArrayList<>();
+
+    public static void addAccessor(Function<AccessibleContext, Object> accessor) {
+        accessors.add(accessor);
+    }
+
+    public static Accessible getAccessible(AccessibleContext context) {
```

Comment:
Will the fix have more code if it will be implemented via SwingAccessor just to access that "private" method?

Reply:
Thank you.
I removed adding accesses for TableCell and ListChildren. If that's what you meant.
I would like to keep the implementation of the Accesser itself, if possible. Because in the future it may come in handy in this form, if in the future you still need to add access to the currentComponent for ListChildren and TableCell.
this approach allows us to make the implementation more generic, in case the potential tree is not inherited from Jtree. It also avoids reflection.

Reply:
Probably we can simplify it further. usually, the SwingAccessor is used to store one accessor per class, which does not need to have an additional list of functions. And if the a11y code needs to call different methods for different types of AccessibleComponent then it uses "instanceof" or role check inside the a11y code. Can we do the same here? It will have the same functionality.

Reply:
Done